src/babel.js

```diff
@@ -79,6 +80,64 @@ export function replaceCssWithCallExpression (
   }
 }
 
+// babel-plugin-styled-components
+// https://github.com/styled-components/babel-plugin-styled-components/blob/37a13e9c21c52148ce6e403100df54c0b1561a88/src/visitors/displayNameAndId.js#L49-L93
+
+const findModuleRoot = (filename) => {
+  if (!filename || filename === 'unknown') {
+    return null
+  }
+  let dir = dirname(filename)
+  if (fs.existsSync(pathJoin(dir, 'package.json'))) {
+    return dir
+  } else if (dir !== filename) {
+    return findModuleRoot(dir)
+  } else {
+    return null
+  }
+}
+
+const FILE_HASH = 'emotion-file-hash'
+const COMPONENT_POSITION = 'emotion-component-position'
+
+const getFileHash = (state) => {
+  const { file } = state
+  // hash calculation is costly due to fs operations, so we'll cache it per file.
+  if (file.get(FILE_HASH)) {
+    return file.get(FILE_HASH)
+  }
+  const filename = file.opts.filename
+  // find module root directory
+  const moduleRoot = findModuleRoot(filename)
+  const filePath = moduleRoot && relative(moduleRoot, filename).replace(pathSep, '/')
+  let moduleName = ''
+  if (moduleRoot) {
+    const packageJsonString = fs.readFileSync(pathJoin(moduleRoot, 'package.json'))
+    if (packageJsonString) {
+      try {
+        moduleName = JSON.parse(packageJsonString).name
+      } catch (e) {}
+    }
+  }
+  const code = file.code
+
+  const fileHash = hashArray([moduleName, filePath, code])
+  file.set(FILE_HASH, fileHash)
+  return fileHash
+}
+
+const getNextId = (state) => {
+  const id = state.file.get(COMPONENT_POSITION) || 0
+  state.file.set(COMPONENT_POSITION, id + 1)
+  return id
+}
+
+const getComponentId = (state) => {
+  // Prefix the identifier with css- because CSS classes cannot start with a number
+  // Also in snapshots with jest-glamor-react the hash will be replaced with an index
+  return `css-${getFileHash(state)}${getNextId(state)}`
```

Comment:
`jest-glamor-react` only replaces classes that have the `css-` prefix. I made it so if a class that starts with `css-` isn't in the rule cache it's just concatenated.